google_analytics/aggregation.py

```diff
@@ -19,12 +41,44 @@ def load_train_test_dataframes(data_dir, x_train_file_name='preprocessed_x_train
     x_test = pd.read_csv(os.path.join(data_dir + x_test_file_name),
                          dtype={"fullVisitorId": str},
                          nrows=nrows_test)
-
+    print('Reducing categories')
+    reduce_categories(x_train, x_test, y_train, selec_top_per, max_cat)
     x_train['date'] = x_train['date'].apply(lambda x: pd.datetime.strptime(str(x), '%Y-%m-%d'))
     x_test['date'] = x_test['date'].apply(lambda x: pd.datetime.strptime(str(x), '%Y-%m-%d'))
     return x_train, y_train, x_test
 
 
+def reduce_categories(x_train, x_test, y_train, selec_top_per, max_cat):
```

Comment:
If I understand correctly, here we only select the values from the categorical that led to higher revenue. What is the motivation for this? My previous work did something similar but targeted values with high predictive power, rather than high spending (I measure predictive power by the relative difference of the target between costumers of the current value and the global average).

Reply:
Mano, thank you for your feedback, very constructive as always.

In your work shown on the load.py file (I hope that we are talking about the same) you propose to use the criteria of `deviation_revenue = category_revenue - global_mean_revenue` to define the predictive power of every category. I agree that this approach is promising. Thus, and because `global_mean_revenue` is constant (please correct me if I am wrong) `deviation_revenue` and  `category_revenue` are equivalent. Therefore, this pull request follows your work. On the other hand, I agree that a binary category such as `is_usd` can add important information to the model. However, I propose to develop this approach (if wanted) on another pull request.